core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -94,7 +93,7 @@ protected TableScan newRefinedScan(TableOperations ops, Table table, Schema sche
     }
 
     if (PLAN_SCANS_WITH_WORKER_POOL && snapshot.dataManifests().size() > 1) {
-      manifestGroup = manifestGroup.planWith(ThreadPools.getWorkerPool());
+      manifestGroup = manifestGroup.planWith(context().planExecutor());
```

Comment:
I think the logic here needs to be slightly different. If `planWith` was called, then the environment option `PLAN_SCANS_WITH_WORKER_POOL` should be ignored.

```java
  if (snapshot.dataManifests().size() > 1 &&
      (PLAN_SCANS_WITH_WORKER_POOL || context().planExecutor() != ThreadPools.getWorkerPool())) {
    manifestGroup.planWith(context().planExecutor());
  }
```